src/main/java/org/spongepowered/server/mixin/core/entity/player/MixinEntityPlayerMP.java

```diff
@@ -0,0 +1,54 @@
+package org.spongepowered.server.mixin.core.entity.player;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.TextComponentTranslation;
+import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.event.SpongeEventFactory;
+import org.spongepowered.api.event.cause.Cause;
+import org.spongepowered.api.event.cause.NamedCause;
+import org.spongepowered.api.event.message.MessageChannelEvent;
+import org.spongepowered.api.event.message.MessageEvent;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.text.channel.MessageChannel;
+import org.spongepowered.api.text.chat.ChatTypes;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.common.SpongeImpl;
+import org.spongepowered.common.interfaces.IMixinCommandSender;
+import org.spongepowered.common.interfaces.IMixinSubject;
+import org.spongepowered.common.interfaces.entity.player.IMixinEntityPlayerMP;
+import org.spongepowered.common.text.SpongeTexts;
+import org.spongepowered.server.chat.ChatFormatter;
+
+import java.util.Optional;
+
+@Mixin(EntityPlayerMP.class)
+public abstract class MixinEntityPlayerMP extends MixinEntityPlayer implements Player, IMixinSubject, IMixinEntityPlayerMP, IMixinCommandSender {
+
+    @Shadow @Final public MinecraftServer mcServer;
+
+    @Override
+    public MessageChannelEvent.Chat simulateChat(Text message) {
+        checkNotNull(message, "message");
+
+        ITextComponent component = new TextComponentTranslation("chat.type.text", SpongeTexts.toComponent(this.getDisplayNameText()), SpongeTexts.toLegacy(message));
```

Comment:
`TextComponentTranslation component = new TextComponentTranslation(....`, saves the casts below